timemachine/integrator.py

```diff
@@ -78,15 +79,59 @@ def __init__(self, force_fxn, masses, temperature, dt, friction):
         # note: per-atom frictions allowed
         self.ca, self.cb, self.cc = np.expand_dims(ca, -1), np.expand_dims(cb, -1), np.expand_dims(cc, -1)
 
-    def step(self, x, v):
+    def _step(self, x, v, eta):
         """Intended to match https://github.com/proteneer/timemachine/blob/37e60205b3ae3358d9bb0967d03278ed184b8976/timemachine/cpp/src/integrator.cu#L71-L74"""
         v_mid = v + self.cb * self.force_fxn(x)
 
-        new_v = (self.ca * v_mid) + (self.cc * np.random.randn(*x.shape))
+        new_v = (self.ca * v_mid) + (self.cc * eta)
         new_x = x + 0.5 * self.dt * (v_mid + new_v)
 
         return new_x, new_v
 
+    def step(self, x, v):
+        return self._step(x, v, np.random.randn(*x.shape))
+
+    def step_deterministic(self, key, x, v):
+        """Return copies x and v, updated by a single timestep. Accepts a PRNGKey for determinism."""
+        return self._step(x, v, jax.random.normal(key, x.shape))
+
+    def _multiple_steps_deterministic(self, key, x, v, n_steps=1000):
```

Comment:
If this is only for testing, perhaps move to the test file?

Reply:
Is it also useful in cases where `self.force_fxn` is not jax transformable, but we wish to use jax keyed rng instead of numpy global rng?

Reply:
> Is it also useful in cases where self.force_fxn is not jax transformable

ah, that's a good point, I think so. With that in mind it might be best to make it "public" and update the docstring to mention this use case

Reply:
Would it be better than to have one implementation that you can manually jax transform if you need it, rather than having two identical implementations where one is transformed and the other isn't?

Reply:
Done in b1b50ce and f6e6c7e